### PR TITLE
examples : switch input_noecho to input_echo to remove negation

### PR DESCRIPTION
**Pull request by deadprogram**
This PR modifies the example to change the boolean `input_noecho` to `input_echo` to remove negation.

I was reading thru the code, and thought it would probably make it easier for further refactoring later.

Thanks to everyone who has been committing to this project!

### CODE REVIEW

**Review by DannyDaemonic — APPROVED** [review]
This looks good. I had to manually merge the `bool input_echo  = true;` line in because it is now preceded by `bool is_antiprompt = false;`.

Everything works the same and it makes the examples easier to read. No downside in that. I can't speak for the developers of the project, but I like this change.

It will need to be rebased before being committed but I approve.

**Comment by deadprogram** [comment]
Pinging maintainers... :smile_cat:

**Comment by DannyDaemonic** [comment]
I'm in favor of this change but unfortunately improving `main` is low priority right now. I've got a couple pending patches for `main` stuck in limbo myself.

I think part of the issue is that those with permissions are so busy iterating over the LLM code that they don't have time to spend on patches that only improve something that's currently working. When I have a patch that fixes something that's broken or helps fix something that's causing problems, it goes through at reasonable speeds, but when my patch just adds a convenience or even fixes something that's being done wrong but isn't actively crashing, it tends to slip through the cracks.

I assume when the ggml and llama stuff has slowed down some more, things like this can be addressed. I can't think of a reason they would prefer using `input_noecho` over `input_echo` (unless there's some precedent or coding pattern it follows that I don't know about), but some of these other patches I've seen seem excessive or just add extra stuff only the author would use. So I think another part of it is just the influx of pull requests and no one really wanting to be the bad guy, and that further lowers the priorities of these subset of pull requests.

They could add a maintainer whose focus is just on `main` and the other code outside of ggml.c and llama.cpp, but even that is something they'd have to evaluate and spend time discussing, which could take away from the momentum they've built with the evaluation and inference development. Don't get me wrong, I do think there's value in maining a more legible and feature complete `main`, but sometimes when you just don't have the time, lower priority things fall by the side.

**Comment by deadprogram** [comment]
Closing, will revisit developer experience improvements in future iterations.

File: examples/main/main.cpp
Comment (ggerganov) [inline examples/main/main.cpp:253]:
```suggestion
    bool input_echo    = true;
```

**Comment by deadprogram** [comment]
@ggerganov thanks for merging.

What do you think about a PR that added a make task to use something like `astyle` to enforce a specific code formatting for the project?

**Comment by ggerganov** [comment]
Automatic style won't work - I really like vertical aligning things and AFAIK no formatter supports this

**Pull request merged**
